src/rebar3_formatter.erl

```diff
@@ -18,27 +18,30 @@
 -spec format(file:filename_all(), module(), opts()) -> result().
 format(File, Formatter, Opts) ->
     AST = get_ast(File),
-    QuickAST = get_quick_ast(File),
     Comments = get_comments(File),
     FileOpts = apply_per_file_opts(File, Opts),
     {ok, Original} = file:read_file(File),
     Formatted = format(File, AST, Formatter, Comments, FileOpts),
     Result = case Formatted of
-               Original ->
-                   unchanged;
-               _ ->
-                   changed
-             end,
-    case maybe_save_file(maps:get(output_dir, FileOpts), File, Formatted) of
-      none ->
-          Result;
-      NewFile ->
-          case get_quick_ast(NewFile) of
-            QuickAST ->
-                Result;
+            Original ->
+                unchanged;
             _ ->
-                erlang:error({modified_ast, File, NewFile})
-          end
+                changed
+            end,
+    OutputDir = maps:get(output_dir, FileOpts),
+    case maybe_save_file(OutputDir, File, Formatted) of
+        none ->
+            Result;
+        NewFile ->
+            check_quick_ast(File, NewFile, Result)
```

Comment:
If `output_dir` is `current`, i.e. if `File == NewFile`… wouldn't this check be bogus?
I mean… we will have replaced the file with the formatted version already… we won't be using the old AST for comparison.